File: LeetCode/982-minimum-increment-to-make-array-unique/minimum-increment-to-make-array-unique.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int moves = 0;
        for(int i = 0; i < n-1; i++){
            if(nums[i] == nums[i+1]){
                nums[i+1] = nums[i+1] + 1;
                moves++;
            }

            if(nums[i+1] < nums[i]){
                moves += (nums[i] - nums[i+1] + 1);
                nums[i+1] = nums[i+1] + ((nums[i] - nums[i+1]) + 1);
            }
        }

        return moves;
    }
};

File: LeetCode/982-minimum-increment-to-make-array-unique/minimum-increment-to-make-array-unique.go
func minIncrementForUnique(nums []int) int {
    sort.Ints(nums)
    moves := 0
    sz := len(nums)

    for i := 0; i < sz - 1; i++ {
        if nums[i] == nums[i+1] {
            nums[i+1] = nums[i+1] + 1
            moves++
        } else if nums[i+1] < nums[i] {
            diff := (nums[i] - nums[i+1]) + 1
            moves += diff
            nums[i+1] = nums[i+1] + diff
        }
    }

    return moves
}